internal/cache/controller_obj_cache.go

```diff
@@ -2523,37 +2527,69 @@ func checkRequiredValuesYaml() bool {
 // validateAndConfigureSeGroup validates SeGroup configuration provided during installation
 // and configures labels on the SeGroup if not present already
 func validateAndConfigureSeGroup(client *clients.AviClient) bool {
-	// Not applicable for NodePort mode / disable route is set as True
+	// Indulge in Marker based SEGroup searching only when user input is empty.
+	// SE Group selection priority
+	// 1. User input
+	seGroupToUse := lib.GetSEGNameEnv()
+
+	// TODO: pagination
+	uri := "/api/serviceenginegroup/?include_name&page_size=100&cloud_ref.name=" + utils.CloudName
```

Comment:
Now that I think about it, you must pick this pagination up in the next PR before release. I suspect the use case we are trying to achieve here will drive 200+ SE Groups per controller.